.github/workflows/cd.yml

```diff
@@ -0,0 +1,42 @@
+name: Continuous Deployment
+
+on:
```

Comment:
Do we really need a separate cd.yml an deploy.yml? Can we not just have ci.yml and also run `yarn pub` there.

Reply:
Not exactly. If you look, I have modified the `ci.yml` to trigger ONLY on PRs opened against `master`, no longer also on pushes to `master`. This is because we do not want to trigger a publish to npm ANYTIME someone opens a PR against master. So we have to create a seperate workflow for when changes are successfully merged into `master`.

The `cd.yml` is what runs now on pushes and/or merges into `master`. Since we only want to deploy the full changes, as is merged in master, it makes sense to separate them this way. So, this way, `ci.yml` is focused on making sure that PRs are good enough to be merged, while `cd.yml` triggers a new release once any such changes are succcessfully merged into main. This also allows us to detect if the changes merged into main include new failing tests or broken code that we couldn't in the PR run.

Reply:
OK makes sense 👍